htdocs/product/stock/class/mouvementstock.class.php

```diff
@@ -159,6 +159,40 @@ public function _create($user, $fk_product, $entrepot_id, $qty, $type, $price =
 		$error = 0;
 		dol_syslog(get_class($this)."::_create start userid=$user->id, fk_product=$fk_product, warehouse_id=$entrepot_id, qty=$qty, type=$type, price=$price, label=$label, inventorycode=$inventorycode, datem=".$datem.", eatby=".$eatby.", sellby=".$sellby.", batch=".$batch.", skip_batch=".$skip_batch);
 
+		// start hook at beginning
+                global $action, $hookmanager;
+                $hookmanager->initHooks(array('mouvementstock'));
+                // Hook of thirdparty module
+                if (is_object($hookmanager)) {
+                    $parameters = array(
+                        'currentcontext'   => 'mouvementstock',
+                        'user'             => &$user,
+                        'fk_product'       => &$fk_product,
+                        'entrepot_id'      => &$entrepot_id,
+                        'qty'              => &$qty,
+                        'type'             => &$type,
+                        'price'            => &$price,
+                        'label'            => &$label,
+                        'inventorycode'    => &$inventorycode,
+                        'datem'            => &$datem,
+                        'eatby'            => &$eatby,
+                        'sellby'           => &$sellby,
+                        'batch'            => &$batch,
+                        'skip_batch'       => &$skip_batch,
+                        'id_product_batch' => &$id_product_batch
+                    );
+                    $reshook    = $hookmanager->executeHooks('_create', $parameters, $this, $action);    // Note that $action and $object may have been modified by some hooks
```

Comment:
Name of hook is not explicit of what it does. 
My understanding is that hook transform input value before calling the stock movement creation, right ?
So i suggest to name the hook function  "replaceDateBeforeStockMovementCreation"

Reply:
The naming is like what i saw in other hook of this type, but you can change to what ever you want.

The goal of this hook is to give a full access to the stock movements, you can modify the parameters or controle the code execution depending on $reshook : 
0 continue
-1 send an error
1 replace the _create function

personally i made a replacement of _create

that just a full access hook , i let you decide for the naming (of course those kind of super hook could be used in many places in a more generic code)